File: test_reg.py
import requests
import json

url = "http://localhost:4000/auth/register"
data = {
    "email": "test@example.com",
    "password": "password123",
    "name": "Test User",
    "role": "buyer"
}

try:
    response = requests.post(url, json=data)
    print(f"Status Code: {response.status_code}")
    print(f"Response: {response.text}")
except Exception as e:
    print(f"Error: {e}")

File: test_reg.js
const http = require('http');

const data = JSON.stringify({
  email: 'test@example.com',
  password: 'password123',
  name: 'Test User',
  role: 'buyer'
});

const options = {
  hostname: 'localhost',
  port: 4000,
  path: '/auth/register',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': data.length
  }
};

const req = http.request(options, res => {
  console.log(`Status Code: ${res.statusCode}`);
  res.on('data', d => {
    process.stdout.write(d);
  });
});

req.on('error', error => {
  console.error(error);
});

req.write(data);
req.end();
